connectors/aws/aws-dynamodb/src/main/java/io/camunda/connector/aws/dynamodb/model/AwsInputDeserializer.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.aws.dynamodb.model;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+
+/**
+ * Custom deserializer for AwsInput. This deserializer is designed for use with newer template
+ * versions incorporating template generation (version 7 onwards). It also maintains compatibility
+ * with older template versions (such as version 6 and below) that do not use template generation.
+ * This dual compatibility is achieved by handling various JSON structures representing different
+ * operation types.
+ */
+public class AwsInputDeserializer extends JsonDeserializer<AwsInput> {
+
+  /**
+   * Deserializes JSON to the appropriate AwsInput subtype. Identifies the specific action type
+   * (e.g., createTable, deleteTable) based on the JSON content and deserializes it into the
+   * corresponding subtype object.
+   *
+   * @param jp JsonParser for reading JSON content.
+   * @param ctxt Deserialization context.
+   * @return Deserialized AwsInput subtype object.
+   * @throws IOException If an issue occurs while reading JSON content.
+   * @throws JsonProcessingException If an error occurs during JSON processing.
+   */
+  @Override
+  public AwsInput deserialize(JsonParser jp, DeserializationContext ctxt)
+      throws IOException, JsonProcessingException {
+    JsonNode node = jp.getCodec().readTree(jp);
+
+    var operationType = determineActionType(node);
+    ObjectMapper mapper = (ObjectMapper) jp.getCodec();
+    return switch (operationType) {
+      case "createTable" -> mapper.convertValue(node, CreateTable.class);
```

Comment:
Could be nice if these literals would be constants and shared/reused in related places like `@TemplateSubType(id = CONSTANT)`